lldb/tools/lldb-dap/JSONUtils.cpp

```diff
@@ -953,9 +953,9 @@ llvm::json::Value CreateThreadStopped(lldb::SBThread &thread,
     } else {
       body.try_emplace("reason", "breakpoint");
       char desc_str[64];
-      uint64_t bp_id = thread.GetStopReasonDataAtIndex(0);
-      uint64_t bp_loc_id = thread.GetStopReasonDataAtIndex(1);
-      snprintf(desc_str, sizeof(desc_str), "breakpoint %" PRIu64 ".%" PRIu64,
+      break_id_t bp_id = thread.GetStopReasonDataAtIndex(0);
```

Comment:
This looks like an implicit cast from uint64_t to int32_t, how safe is this?

Reply:
We have an abstract API to access stop reason data via `uint64_t SBThread::GetStopReasonDataAtIndex(...)` which returns a `uint64_t`, but the `break_id_t` is defined as `typedef int32_t break_id_t;`. If we want to show the breakpoint ID correctly as a negative number for internal breakpoints, then this should be ok.

Reply:
These are implicitly cast here https://github.com/llvm/llvm-project/blob/e823136d43c40b0a9ba6930fd285768f1b46fcb6/lldb/source/API/SBThread.cpp#L251C35-L251C48 to uint64_t, this is converting it back to the internal type. This should be okay.